From b7c2cd2ed38e28ac0c88652e0ef7c93aaada7b33 Mon Sep 17 00:00:00 2001
From: Chen Liangjun <b36089@freescale.com>
Date: Fri, 16 Nov 2012 15:21:08 +0800
Subject: [PATCH 366/505] ENGR00233915 ASRC TEST: add taill for output buffer
 to get all data generated

When convert 8k to 48k within input buffer is 100 bytes, it is possible
that convert data after ASRC processing is 601. If delete the 1 data,
user would hear noise.

In this patch, add tail buffer appended into caculated output buffer to
receive all data generated by ASRC.

Signed-off-by: Chen Liangjun <b36089@freescale.com>
---
 test/mxc_asrc_test/mxc_asrc_test.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/test/mxc_asrc_test/mxc_asrc_test.c b/test/mxc_asrc_test/mxc_asrc_test.c
index f2f132b..4f54017 100644
--- a/test/mxc_asrc_test/mxc_asrc_test.c
+++ b/test/mxc_asrc_test/mxc_asrc_test.c
@@ -184,12 +184,16 @@ int play_file(int fd_asrc, struct audio_info_s *info)
 	char *input_p;
 	char *output_p;
 	int output_dma_size;
+	unsigned int tail;
 
 	input_p = (char *)input_buffer;
 	output_p = (char *)output_buffer;
 	output_dma_size =
 	    asrc_get_output_buffer_size(DMA_BUF_SIZE, info->sample_rate,
 					info->output_sample_rate);
+
+	tail = info->channel * 4 * 4;
+
 	convert_flag = 1;
 	memset(input_null, 0, DMA_BUF_SIZE);
 	if ((err = ioctl(fd_asrc, ASRC_START_CONV, &pair_index)) < 0)
@@ -210,7 +214,7 @@ int play_file(int fd_asrc, struct audio_info_s *info)
 			buf_info.input_buffer_length = DMA_BUF_SIZE;
 		}
 
-		buf_info.output_buffer_length = output_dma_size;
+		buf_info.output_buffer_length = output_dma_size + tail;
 		buf_info.output_buffer_vaddr = output_p;
 		if ((err = ioctl(fd_asrc, ASRC_CONVERT, &buf_info)) < 0)
 			goto error;
-- 
1.8.0

